File: legal-docs-assistant/ingest.py
import asyncio
import os
import shlex
from pathlib import Path

from dotenv import load_dotenv

from supermemory_bash import create_bash


DOCS_DIR = Path(__file__).parent / "docs"
CONTAINER_PATH = "/contracts"


async def main() -> None:
    load_dotenv()

    result = await create_bash(
        api_key=os.environ["SUPERMEMORY_API_KEY"],
        container_tag="legal_docs",
    )
    bash = result.bash

    await bash.exec(f"mkdir -p {CONTAINER_PATH}")

    files = sorted(DOCS_DIR.glob("*.txt"))
    if not files:
        print(f"No .txt files found in {DOCS_DIR}")
        return

    for path in files:
        content = path.read_text()
        target = f"{CONTAINER_PATH}/{path.name}"
        cmd = f"cat > {shlex.quote(target)} <<'SMEOF_DOC'\n{content}\nSMEOF_DOC"
        r = await bash.exec(cmd)
        if r.exit_code != 0:
            print(f"FAILED {path.name}: {r.stderr}")
            continue
        print(f"ingested {target} ({len(content)} bytes)")

    listing = await bash.exec(f"ls -la {CONTAINER_PATH}")
    print("\nContainer contents:")
    print(listing.stdout)


if __name__ == "__main__":
    asyncio.run(main())

File: docs-answering-agent/ingest.ts
import "dotenv/config";
import { readFileSync, readdirSync } from "node:fs";
import { join } from "node:path";
import { createBash } from "@supermemory/bash";

function shellQuote(s: string): string {
  return `'${s.replace(/'/g, `'\\''`)}'`;
}

async function main() {
  const apiKey = process.env.SUPERMEMORY_API_KEY;
  if (!apiKey) {
    throw new Error("SUPERMEMORY_API_KEY is not set");
  }

  const { bash } = await createBash({
    apiKey,
    containerTag: "docs_agent",
  });

  const docsDir = join(process.cwd(), "docs");
  const files = readdirSync(docsDir).filter((f) => f.endsWith(".md"));

  if (files.length === 0) {
    console.log("No markdown files found in docs/");
    return;
  }

  await bash.exec("mkdir -p /docs");

  for (const file of files) {
    const content = readFileSync(join(docsDir, file), "utf8");
    const target = shellQuote(`/docs/${file}`);
    const result = await bash.exec(
      `cat > ${target} <<'__SM_EOF__'\n${content}\n__SM_EOF__`,
    );
    if (result.exitCode !== 0) {
      console.error(`Failed to write ${file}: ${result.stderr}`);
      continue;
    }
    console.log(`Ingested /docs/${file} (${content.length} bytes)`);
  }

  const ls = await bash.exec("ls -la /docs");
  console.log("\nContainer contents:");
  console.log(ls.stdout);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
